docs/conf.py

```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
```

Comment:
why dont you also include https://github.com/emdgroup/baybe/discussions/64 and https://github.com/emdgroup/baybe/discussions/65 into this PR?

Reply:
Also thought about that and am currently testing possible solutions for this :)
Original reason was that I wanted a "quick fix" for just making the docs pass

Reply:
Also, order of the examples would now look like this:
![image](https://github.com/emdgroup/baybe/assets/84016733/a2e69023-f051-4f6d-99e2-00a1576cbcc4)

Reply:
This is now the ordering of the examples:
![image](https://github.com/emdgroup/baybe/assets/84016733/439c8ee1-13a3-40e5-be83-b3730d66fb01)
So once we aligned on the how many levels the headers should have, I'll push everything and we are gucci to merge

Reply:
Resolved as this was discussed in a meeting.